src/parse/decl.rs

```diff
@@ -209,7 +210,7 @@ impl<I: Iterator<Item = Lexeme>> Parser<I> {
         self.expect(Token::Comma)?;
         loop {
             let decl = self
-                .declarator(false)?
+                .declarator(false, first_qualifiers)?
                 .expect("declarator should return Some when called with allow_abstract: false");
             let location = decl.id().unwrap().location;
             let (id, ctype) = decl
```

Comment:
This should also check that `inline` isn't set. From 6.7.4  Function specifiers, paragraph 2:

> Function specifiers shall be used only in the declaration of an identifier for a function.